mesop/dataclass_utils/dataclass_utils.py

```diff
@@ -54,13 +75,29 @@ def dataclass_with_defaults(cls: Type[C]) -> Type[C]:
       elif get_origin(type_hint) == dict:
         setattr(cls, name, field(default_factory=dict))
       elif isinstance(type_hint, type):
-        setattr(
-          cls, name, field(default_factory=dataclass_with_defaults(type_hint))
-        )
-
+        if has_parent(type_hint):
+          # If this isn't a simple class (i.e. it inherits from another class)
+          # then we will preserve its semantics (not try to set default values
+          # because it's not a dataclass) and instantiate it with each new instance
+          # of a state class.
+          setattr(cls, name, field(default_factory=type_hint))
```

Comment:
What about the case where the constructor requires that certain parameters be set? I'm assuming an exception would be thrown in that case?

Reply:
yeah - an exception will be thrown - if certain parameters are required, then they will need to write a custom default factory.